src/apis/axios.ts

```diff
@@ -1,37 +1,70 @@
-import axios from 'axios';
+import axios from "axios";
+import type { AxiosRequestConfig, AxiosError } from "axios";
+import { requestRefreshToken } from "@/apis/auth";
+import queryString from "query-string";
 
-const instance = axios.create({
+interface RetryConfig extends AxiosRequestConfig {
+  _retry?: boolean;
+}
+
+const paramsSerializer = (params: any) =>
+  queryString.stringify(params, { skipEmptyString: true, skipNull: true });
```

Comment:
```suggestion
const paramsSerializer = (params: Record<string, any>) =>
  queryString.stringify(params, { skipEmptyString: true, skipNull: true });
```

```ts
export function stringify(
  // TODO: Use the below instead when the following TS issues are fixed:
  // - https://github.com/microsoft/TypeScript/issues/15300
  // - https://github.com/microsoft/TypeScript/issues/42021
  // Context: https://github.com/sindresorhus/query-string/issues/298
  // object: StringifiableRecord,
  object: Record<string, any>,
  options?: StringifyOptions
): string;
```
query-string의 stringify은 객체를 인자로 받기 때문에 paramsSerializer도 객체로 인자를 받게끔 정의하는게 좋습니다.
객체 타입을 허용하는 타입으로 Record를 많이 사용합니다.